src/org/zaproxy/zap/extension/autoupdate/AddOnsTableModel.java

```diff
@@ -109,10 +123,12 @@ protected AddOn getAddOnForDownload(AddOnWrapper aow) {
     protected void addAddOnWrapper(AddOn addOn, AddOnWrapper.Status status) {
         AddOnWrapper aow = createAddOnWrapper(addOn, status);
         int idx = 0;
-        for (; idx < getAddOnWrappers().size(); idx++) {
-            if (comparator.compare(aow, getAddOnWrappers().get(idx)) < 0) {
-                break;
-            }
+    	if (comparator != null) {
+    		for (; idx < getAddOnWrappers().size(); idx++) {
+        		if (comparator.compare(aow, getAddOnWrappers().get(idx)) < 0) {
+        			break;
+        		}
+        	}
         }
 
         getAddOnWrappers().add(idx, aow);
```

Comment:
This statement should be moved to the above if-then block, and in the (new) else block simply add the wrapper:
```Java
idx = getAddOnWrappers().size();
getAddOnWrappers().add(aow);
```
it avoids copying arrays around when there's no comparator, which should be the case from now on.